### ISSUE TITLE
Limit imports only to symbols in __init__.py

### ISSUE THREAD

**Issue by hgrecco** [seddonym/import-linter#123]
Awesome package. I was about to build something until I found this! Congrats

I have a something that looks like this
```
- __init__.py
- a.py
- b.py
- c.py
- inner
  - __init__.py
  - inner1
    - __init__.py 
    - a1.py
    - b1.py
    - c1.py
  - inner2
    - __init__.py
    - a2.py
    - b2.py
    - c2py
```

and I am trying to restrict imports from inner1 and inner2 only to those imported in their respective `__init__.py`  How can I achieve this?

**Comment by seddonym** [comment]
Thanks for raising the issue!

I'm not totally clear on what you are trying to restrict, could you give an example?

**Comment by seddonym** [comment]
Interesting! I think there is a lot of mileage in the idea of private modules - in fact we use a (slightly different) convention at work where modules prefixed with an underscore are deemed 'private', and can only be imported by their parent `__init__.py` file. At some point I plan to make a built in contract type for this.

I think the best way forward with your requirements is to create a [custom contract](https://import-linter.readthedocs.io/en/stable/custom_contract_types.html). Let me know if anything in the documentation isn't clear.

**Comment by hgrecco** [comment]
Thanks! , I will take a look!

**Comment by seddonym** [comment]
> To me by default everything is private and only things imported in main init file are public, so, nobody should do from a.xxx import yyy, but do from a import yyy.

We have a similar convention at my workplace, where we prefer importing modules rather than objects within the modules.

I think this would be most suited to its own contract type. It's not something I'm likely to prioritize at the moment, but I think would be possible for someone to create [a custom contract type](https://import-linter.readthedocs.io/en/stable/custom_contract_types.html) that does this, and release that as a separate Pypi package if they wanted it to be useable by others.

**Comment by seddonym** [comment]
Makes sense. That sounds like a convention you might want to adopt - but other conventions may be just as good. I don't think Import Linter should be opinionated about conventions like that though.
